src/routers/user.ts

```diff
@@ -0,0 +1,80 @@
+import express from 'express';
+import { UserService } from '../services/userService';
+import {
+  userBodyValidatorOnCreate,
+  userBodyValidatorOnUpdate,
+  userParamsIdValidator,
+  userQuerySubstringLimitValidator,
+} from '../validation/validators';
+import { User } from '../types';
+
+const user = express.Router();
+const userService = new UserService();
+
+user.post('/user', userBodyValidatorOnCreate, async (req, res) => {
+  const user: User = { ...req.body };
+  const result = await userService.createUser(user);
+  if (result instanceof Error) {
+    res.status(500).json({ message: result });
+  } else {
+    res.status(200).json({ createdUser: result });
+  }
+});
+
+user.get('/user/:id', userParamsIdValidator, async (req, res) => {
+  const { id } = req.params;
+  const result = await userService.getUserById(id);
+  if (result instanceof Error) {
+    res.status(500).json({ message: result });
+  } else {
+    result
+      ? res.status(200).json({ user: result })
+      : res.status(404).json({ message: `no users with id ${id}` });
+  }
+});
+
+user.put(
+  '/user/:id',
+  userBodyValidatorOnUpdate,
+  userParamsIdValidator,
+  async (req, res) => {
+    const { id } = req.params;
+    const result = await userService.updateUser(id, req.body);
+    if (result instanceof Error) {
+      res.status(500).json({ message: result });
+    } else {
+      result
+        ? res.status(200).json({ updatedUser: result })
+        : res.status(404).json({ message: `no users with id ${id}` });
+    }
+  }
+);
+
+user.delete('/user/:id', userParamsIdValidator, async (req, res) => {
+  const { id } = req.params;
+  const result = await userService.deleteUser(id);
+  if (result instanceof Error) {
+    res.status(500).json({ message: result });
+  } else {
+    result
+      ? res
+          .status(200)
+          .json({ message: `user with id ${id} marked as deleted` })
+      : res.status(404).json({ message: `no users with id ${id}` });
+  }
+});
+
+user.get('/users', userQuerySubstringLimitValidator, async (req, res) => {
```

Comment:
I missed pointing out this in the previous module. When you use validation middleware, you'd better make your req type more specific (see below)
![image](https://user-images.githubusercontent.com/70232156/212261723-a98a274f-4088-460f-b5e3-199be5c33b2a.png)

In this case you will get concrete types from your params or queryparams. Please, look this up.